### ISSUE TITLE
razzle-start-server-webpack-plugin: monitor.js -> Cannot find update.

### ISSUE THREAD

**Issue by Bram-Zijp** [jaredpalmer/razzle#1579]
## 🐛 Bug report
Not sure if this is a bug or a feature.

I was using v3.4.5 with the experimental features all enabled. Moving to v4.0.3 made the re-compilation of components (on change), real slow (it's a big project). It happens with or without `react-refresh` This is likely because of the new way of hot reloading as presented in `razzle-start-server-webpack-plugin/src/monitor.js`.

### Current Behavior
Rebuilds unrelated modules.

### Expected behavior
Rebuilding only changed modules.

### Reproducible example

I will try to attempt to reproduce it in a fresh project.

### Suggested solution(s)

Not sure if it's a bug or a feature. Would be awesome if you could provide an answer which explains why this message is shown in the latest but not in 3.4.5. Like I'm very curious about the motivation around this way of handling updates.
### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Razzle           |  4.0.3 |
| Razzle Plugins   | css |
| Node             | 12.20.0 |
| Browser          | 
| npm/Yarn         | npm @ 6.14.8 |
| Operating System | Linux/Ubuntu/PopOS |
| TypeScript       | 3.8.3
| React            | 16.8.6

**Comment by fivethreeo** [comment]
It is a new version of start-server plugin. May be a bug, HMR is finnicky.. Not sure how to fix this. :(

**Comment by fivethreeo** [comment]
maybe hot is not set.. hm..

**Comment by Bram-Zijp** [comment]
I feel like it might have to do with anonymous components (not sure if this is true as I get the same behavior without React Refresh).

I noticed that with 1 project @ razzle 3.4.5 hot reloads properly (a rather fresh project, soft page reloads). The project I upgraded from 3.4.5 to 4.0.3 is rather old and probably uses a lot of anonymous components. The old project is hard reloading pages on change @ razzle 3.4.5 (it does seem to compile faster as @ razzle 4.0.3).

I notice Next.js has a codemod to automatically name anonymous components (I don't use Next.js though and I doubt there is something for a default CRA/Webpack setup): 
https://nextjs.org/docs/advanced-features/codemods#name-default-component

Thinking about it longer, it should not be a problem of Razzle. I do think that more modules get re-compiled with the razzle monitor module.hot implementation. Could also be that the thread-loader/babel-loader cache is not working as performant. 

I feel like this is not an issue in the Razzle module but I will log my findings here.

**Comment by Bram-Zijp** [comment]
Given the project isn't setup on a react-refresh proof way, I ended up disabling react refresh. Also I ensured only that only thread-loader + babel-loader are ran on the TypeScript files. Using this in combination with webpack 5 long term caching ended up to give a pretty similar recompile performance. As a bonus, start up time is way faster and bundle-analyzer works better (not sure if this is because of Webpack 5)
